app/controllers/api/v1/restaurants_controller.rb

```diff
@@ -69,9 +74,16 @@ def filter_restaurants(open_street_map_restaurants, search)
         if restaurant_candidate.save
           restaurants.push(restaurant_candidate)
         end
-
       end
     end
     return restaurants
   end
+
+  protected
+  def authenticate_user
+    if !user_signed_in?
+      render json: {error: ["You need to be signed in first"]}
+    end
+  end
```

Comment:
This method is being used by all of your controllers, it should be moved to the parent `ApplicationController` that all of these controllers inherit from